chi/network.py

```diff
@@ -617,15 +626,14 @@ def get_free_floating_ip() -> dict:
 
     Returns:
         The free floating IP representation.
-
-    Raises:
-        ValueError: If your project has no free floating IPs.
     """
-    ips = list_floating_ips()
-    unbound_fip = next(iter([ip for ip in ips if ip['port_id'] is None]), None)
-    if not unbound_fip:
-        raise ValueError('No free floating IP found')
-    return unbound_fip
+    ips = neutron().list_floatingips()['floatingips']
+    unbound = (ip for ip in ips if ip['port_id'] is None)
+    try:
+        fip = next(unbound)
+        return fip
+    except StopIteration:
+        print("No free floating IP found")
```

Comment:
This is a bit simpler than catching StopIteration; the `next` takes a default param.

```suggestion
    fip = next(unbound, None)
    if not fip:
        print("No free floating IP found")
    return fip
```

Why would we not raise an exception here, though? Consider that there are users of this library not interactively using Jupyter. For example, I have some scripts that may do this:

```python
from chi.network import get_free_floating_ip

try:
    fip = get_free_floating_ip()
except:
    LOG.info("Could not allocate floating IP, doing something else...")
    pass
```

This change will break that usage and will thus be a breaking change. I would recommend that the tutorial/whatever that is calling this function instead surround its call with a `try/except`; that's the way to handle such things IMO.

Reply:
I incorporated this but am not super handy with Python exceptions, so am leaving this conversation open for now.